File: app/rotas/participante.py
from app.rotas import *
from app.services.participante import *

def registerParticipante(app):
    
    @app.route('/participante/buscar', methods=['GET'])
    def retornarParticipantes():
        tokenSala = request.args.get('tokenSala')

        arrNomes = retornarNomesParticipantes(tokenSala)

        return jsonify({'data':arrNomes})
    
    @app.route('/participante/retornarAmigo', methods=['GET'])
    def retornarAmigo():
        meuNome = request.args.get('meuNome')
        token = request.args.get('tokenSala')

        response = retornarAmigoSecreto(token,meuNome)
        
        return jsonify(response)
    
    @app.route('/participante/retornarSorteio', methods=['GET'])
    def retornarSorteio():
        token = request.args.get('tokenSala')

        data = retornarSorteioDB(token)

        return jsonify({"revelacao":data})
    
    @app.route('/participante/teste', methods=['GET'])
    def rotaTeste():
        data = {'msg':'ok'}
        return jsonify(data)

File: app/static/js/routes/participante.js
// requisição para retornar os participantes da sala expecifica
export async function retornarParticipantes(tokenSala) {

    var response = axios.get('/participante/buscar', {params:{tokenSala:tokenSala}})

    return response
    
}

// requisição para retornar a pessoa de oração
export async function retornarAmigoSecreto(meuNome, tokenSala) {

    var response = axios.get('/participante/retornarAmigo', {params:{meuNome:meuNome,tokenSala:tokenSala}})

    return response
    
}

// requisição para retornar o sorteio da sala especifica
export async function retornarSorteio(tokenSala) {

    var response = axios.get('/participante/retornarSorteio', {params:{tokenSala:tokenSala}})

    return response
    
}